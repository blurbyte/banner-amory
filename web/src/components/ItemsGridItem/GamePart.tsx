// number of game part with banner background

import * as React from 'react';
import styled from 'styled-components';

import { Item } from '@sharedTypes/Item';
import { Colors } from '@styles/colors';
import { Banner } from '../Icons';
import Rank from './Rank';

const Wrapper = styled.div`
  width: 2rem;
  color: ${Colors.red};
  position: absolute;
  z-index: 9;
  top: 0;
  right: 0;
`;

const Part = styled(Rank)`
  top: 0.4rem;
  background: transparent;
  color: ${Colors.white};
  padding-bottom: 0;
`;

type GamePartProps = {
  gamePart: Item['gamePart'];
};

class GamePart extends React.Component<GamePartProps> {
  render() {
    const { gamePart } = this.props;

    return (
      <Wrapper>
        <Part>{gamePart}</Part>
        <Banner />
      </Wrapper>
    );
  }
}

export default GamePart;
