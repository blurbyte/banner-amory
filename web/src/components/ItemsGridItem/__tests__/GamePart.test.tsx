import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import GamePart from '../GamePart';

jest.mock('../../Icons', () => ({
  Banner: 'Banner'
}));

test('renders correctly', () => {
  const component = renderer.create(<GamePart gamePart={2} />);
  expect(component.toJSON()).toMatchSnapshot();
});
