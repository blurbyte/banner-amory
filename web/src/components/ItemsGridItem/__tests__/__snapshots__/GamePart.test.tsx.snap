// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  right: 0;
  bottom: 0;
  z-index: 99;
  background: #ffffff;
  margin: 0;
  padding: 0;
  width: 2rem;
  height: 2rem;
  font-weight: 600;
  padding-bottom: 0.2rem;
}

.c0 {
  width: 2rem;
  color: #d92f29;
  position: absolute;
  z-index: 9;
  top: 0;
  right: 0;
}

.c1 {
  top: 0.4rem;
  background: transparent;
  color: #ffffff;
  padding-bottom: 0;
}

<div
  className="c0"
>
  <p
    className="c1 c2"
  >
    2
  </p>
  <Banner />
</div>
`;
